numpy/core/src/umath/loops.c.src

```diff
@@ -2346,8 +2350,18 @@ HALF_floor_divide(char **args, npy_intp const *dimensions, npy_intp const *steps
     BINARY_LOOP {
         const npy_half in1 = *(npy_half *)ip1;
         const npy_half in2 = *(npy_half *)ip2;
-        npy_half mod;
-        *((npy_half *)op1) = npy_half_divmod(in1, in2, &mod);
+
+        float fh1 = npy_half_to_float(in1);
+        float fh2 = npy_half_to_float(in2);
+        float div, modf;
+
+        if (!fh2) {
+            div = fh1 / fh2;
+            *((npy_half *)op1) = npy_float_to_half(div);
+        } else {
+            div = npy_divmodf(fh1, fh2, &modf);
+            *((npy_half *)op1) = npy_float_to_half(div);
```

Comment:
i know this is a bit uglier than before, but had to move the `npy_half_divmod` to these functions, to make float16 work.

Reply:
This is likely caused by the clang compiler on mac os  likely because clang doesnt implement annex F of C standard, I think this is still open. (https://bugs.llvm.org/show_bug.cgi?id=17005)

Reply:
Not sure we have to work around compiler shortcomings/bugs, but its also not wrong to do. We had similar issues with windows±SSE before for example, IIRC.

If you want to skip tests, I think `platform.python_compiler()` should probably work here?

Reply:
thank you Sebastian, for your inputs ! i think i will skip the test for now. 

> Note that NumPy is developed mainly using GNU compilers. Compilers from other vendors such as Intel, Absoft, Sun, NAG, Compaq, Vast, Portland, Lahey, HP, IBM, Microsoft are only supported in the form of community feedback, and may not work out of the box. GCC 4.x (and later) compilers are recommended.

This is what the devdocs (https://numpy.org/devdocs/user/building.html) says: do we want to change this given that we now test on clang  in the CI ?

Reply:
On macOS, clang is the default compiler. On windows we build with MSVC.

Reply:
@mattip should i change the dev docs to reflect that ?

Reply:
Yeah, I think we should mention clang and msvc as major compilers there, intel maybe too, but I am not sure. I am not even sure GCC 4.x is still reasonably new to be honest.

Reply:
yes gcc 4.x is pretty old, but since we still use it in our CI good to mention 4.x and later in docs.

Reply:
> Not sure we have to work around compiler shortcomings/bugs,

If we want the error to appear reliably, we should be using one of the `npy_set_floatstatus_` variants rather than skipping the test. If we don't care, then skipping the test is fine.

Reply:
Right, maybe that is better and yeah, we have done that in a few places. Would probably be good to put it behind a compiler directive to only use it when necessary. (Its too bad we won't notice when it gets fixed, unless we go through the trouble and put it into the conftests...).

Reply:
yes that sounds like a good idea. so if i understand correctly:i should use #ifdef __clang__  wherever i use 1.0/0.0 in my code currently to set status and continue with what I currently have for non clang compilers. I am wondering if the npy_set_floatstatus_ work for clang though ? also if we are going to call this flag, what do we set these values to if not nan and infs (the NPY_INFINITY and NPY_NAN i doubt can be used on all platforms.)